collectors/payments_collector.go

```diff
@@ -0,0 +1,153 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	totalPayments = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "lnd_total_payments",
+			Help: "Total number of payments initiated",
+		},
+	)
+	totalHTLCAttempts = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "lnd_total_htlc_attempts",
+			Help: "Total number of HTLC attempts across all payments",
+		},
+	)
+	paymentAttempts = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "lnd_payment_attempts_per_payment",
+			Help:    "Histogram tracking the number of attempts per payment",
+			Buckets: prometheus.ExponentialBucketsRange(1, 2, 10),
+		},
+	)
+)
+
+// paymentsMonitor listens for payments and updates Prometheus metrics.
+type paymentsMonitor struct {
+	client routerrpc.RouterClient
+
+	ctx context.Context
```

Comment:
This is a bit of an anti-pattern to have it as a member, maybe the context could be created at `start` time with `LndServices` as a field. In general there are many occurrences of `ctx.Background` in this repo, which needs an overhaul as well (not this PR).

Reply:
Updated. Let me know if it looks good.

Reply:
Looks good, a nice pattern would be to use a `fn.ContextGuard`, but since that's not used in this repo for other structs, it's better to keep things consistent.

Reply:
Sounds good. I'll check the type out and keep it in mind going forward. Thanks for the tip.